ClassTranscribeServer/Controllers/OfferingsController.cs

```diff
@@ -37,36 +37,38 @@ public async Task<ActionResult<IEnumerable<OfferingDTO>>> GetOfferingsByStudent(
             // Store the results
             List<Offering> offerings = await _context.Offerings.ToListAsync();
 
-
-            // Filter out offerings where there is no media items available.
-            var filteredOfferings = offerings
+            // Filter out offerings where there is no media items available
+            return offerings
                 .FindAll(o => o.Playlists != null && o.Playlists.SelectMany(m => m.Medias).Any())
                 .OrderBy(o => o.Term.StartDate)
+                .Select(o => GetOfferingDTO(o))
                 .ToList();
+        }
 
-            var offeringListDTO = filteredOfferings.Select(o => new OfferingDTO
+        // GET: api/Offerings/ByInstructor/{userId}
+        /// <summary>
+        /// Gets all offerings for an instructor
+        /// </summary>
+        [HttpGet("ByInstructor/{userId}")]
+        [Authorize(Roles = Globals.ROLE_ADMIN + "," + Globals.ROLE_INSTRUCTOR)]
```

Comment:
So this line is required in addition to the explicit check below? Asking out of ignorance

Reply:
Yes. This line allows admins + instructors to use this endpoint (and immediately denies any normal or non-logged-in user), whereas the check below is for explicitly checking whether the passed-in instructorID is the same as the current logged-in user's ID (since we only want to allow people to view this data about themselves) while allowing admins to view any instructor's data.